### PR TITLE
Firecloud Auth

### PR DESCRIPTION
**Pull request by Horneth**
Fix firecloud auth and pass the whole `JesWorkflowPaths` around in the `BackendInitializationData`

### CODE REVIEW

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesWorkflowPaths.scala [hunk 1]
```diff
@@ -13,22 +17,36 @@ object JesWorkflowPaths {
...
 
-class JesWorkflowPaths(workflowDescriptor: BackendWorkflowDescriptor, jesConfiguration: JesConfiguration, gcsFileSystem: GcsFileSystem) {
+  implicit class GoogleAuthWorkflowOptions(val workflowOptions: WorkflowOptions) extends AnyVal {
+    def toGoogleAuthOptions: GoogleAuthMode.GoogleAuthOptions = new GoogleAuthOptions {
+      override def get(key: String): Try[String] = workflowOptions.get(key)
+    }
+  }
+}
 
+class JesWorkflowPaths(workflowDescriptor: BackendWorkflowDescriptor, jesConfiguration: JesConfiguration, val gcsFileSystem: GcsFileSystem) {
```
Comment (mcovarr) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesWorkflowPaths.scala]: Maybe `gcsFileSystem` should get a more specific name to prevent the confusion that led to the original bug?  `gcsFileSystemWithUserAuth` or something?

Comment (Horneth) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesWorkflowPaths.scala]: Good point

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesWorkflowPaths.scala [hunk 2]
```diff
@@ -13,22 +17,36 @@ object JesWorkflowPaths {
   def apply(workflowDescriptor: BackendWorkflowDescriptor, jesConfiguration: JesConfiguration, gcsFileSystem: GcsFileSystem) = {
     new JesWorkflowPaths(workflowDescriptor, jesConfiguration, gcsFileSystem)
   }
-}
 
-class JesWorkflowPaths(workflowDescriptor: BackendWorkflowDescriptor, jesConfiguration: JesConfiguration, gcsFileSystem: GcsFileSystem) {
+  implicit class GoogleAuthWorkflowOptions(val workflowOptions: WorkflowOptions) extends AnyVal {
```
Comment (mcovarr) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesWorkflowPaths.scala]: Can't this use the copy of this implicit class in the package object?  We have three copies of this already... 😢

**Comment by mcovarr** [comment]
:+1:

**Comment by geoffjentry** [comment]
@Horneth This is what you had to do to unstuck @helgridly, right?

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesCallPaths.scala [hunk 3]
```diff
@@ -13,7 +13,7 @@ object JesCallPaths {
 }
 
 class JesCallPaths(jobKey: BackendJobDescriptorKey, workflowDescriptor: BackendWorkflowDescriptor, jesConfiguration: JesConfiguration,
-                   val gcsFileSystem: GcsFileSystem) extends JesWorkflowPaths(workflowDescriptor, jesConfiguration, gcsFileSystem) {
+                   gcsFileSystem: GcsFileSystem) extends JesWorkflowPaths(workflowDescriptor, jesConfiguration, gcsFileSystem) {
```
Comment (geoffjentry) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesCallPaths.scala]: Not a quibble, just curious why the change

**Comment by geoffjentry** [comment]
👍

**Comment by Horneth** [comment]
@geoffjentry correct

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesCallPaths.scala (hunk 3 above)
Comment (Horneth) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesCallPaths.scala]: Because I made it a `val` in `JesWorkflowPaths`, which `JesCallPaths` extends from, so there was no need to have it here (I actually think I remember getting an error for not overriding it or something).
